source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -421,11 +427,99 @@ void SetupContextForAmazonServiceAccount(string @namespace, string clusterUrl, s
                 var clusterName = variables.Get(SpecialVariables.EksClusterName);
                 log.Info($"Creating kubectl context to {clusterUrl} (namespace {@namespace}) using EKS cluster name {clusterName}");
 
+                if (TrySetKubeConfigAuthenticationToAwsCli(clusterName, clusterUrl, user))
+                {
+                    return;
+                }
+
+                log.Verbose("Attempting to authenticate with aws-iam-authenticator");
+                SetKubeConfigAuthenticationToAwsIAm(user, clusterName);
+            }
+
+            bool TrySetKubeConfigAuthenticationToAwsCli(string clusterName, string clusterUrl, string user)
+            {
+                log.Verbose("Attempting to authenticate with aws-cli");
+                if (TrySetAws())
+                {
+                    try
+                    {
+                        var awsCliVersion = GetAwsCliVersion();
+                        var minmumAwsCliVersionForAuth = new Version("1.16.156");
+                        if (awsCliVersion > minmumAwsCliVersionForAuth)
+                        {
+                            var region = GetEksClusterRegion(clusterUrl);
+                            if (!string.IsNullOrWhiteSpace(region))
+                            {
+                                var apiVersion = GetEksClusterApiVersion(clusterName, region);
+                                SetKubeConfigAuthenticationToAwsCli(user, clusterName, region, apiVersion);
+                                return true;
+                            }
+                            
+                            log.Verbose("The EKS cluster Url specified should contain a valid aws region name");
+                        }
+
+                        log.Verbose($"aws cli version: {awsCliVersion} does not support the \"aws eks get-token\" command. Please update to a version later than 1.16.156");
+                    }
+                    catch (Exception e)
+                    {
+                        log.Verbose($"Unable to authenticate to {clusterUrl} using the aws cli. Failed with error message: {e.Message}");
+                    }
+                }
+                else
+                {
+                    log.Verbose("Could not find the aws cli, falling back to the aws-iam-authenticator.");
+                }
+
+                return false;
+            }
+            
+            string GetEksClusterRegion(string clusterUrl) => clusterUrl.Replace(".eks.amazonaws.com", "").Split('.').Last();
+
+            Version GetAwsCliVersion()
```

Comment:
I think the `Version` here is a `System.Runtime.Version`, which is used to represent [the version number of an assembly, operating system, or the common language runtime](https://learn.microsoft.com/en-us/dotnet/api/system.version?view=net-6.0). In short, it is a .net specific thing, and the format is in `major.minor.build.revision`, which differs from SemVer. The outcome that it also works for SemVers in our program is a coincident.

I think we should directly use the `SemanticVersion` class. IMHO correct mapping of the version parts is more important than the ability to use the overloaded `<` operator for readability.